extensions/answer_summarizers/models.py

```diff
@@ -43,42 +43,79 @@
-                for (key, val) in value.iteritems()]))
-    else:
-        return value
-
-
-def _count_answers(answer_dicts_list):
-    """Count an input list of answer objects using collections.Counter. This
-    returns a list of pairs with the first element being an answer object and
-    the second being the number of times it shows up in the input list.
+
+    @classmethod
+    def _hashable(cls, value):
+        """This function returns a hashable version of the input value.
+
+        It converts the built-in collections into their hashable counterparts
+        {list: (tuple), set: (frozenset), dict: (sorted tuple of pairs)}.
+        Additionally, their elements are converted to hashable values through
+        recursive calls. All other values are assumed to already be hashable.
+
+        For single-element lists and sets, the single element is returned by
+        itself.
+        """
+        if isinstance(value, list):
+            # Can't be sure that all elements of lists are hashable, so apply
+            # logic recursively to each element.
+            if len(value) == 1:
+                # Single-elements can be returned by themselves.
+                return cls._hashable(value[0])
+            else:
+                return tuple(cls._hashable(e) for e in value)
+        elif isinstance(value, set):
+            # Set elements are always hashable; don't need to call recursively.
+            if len(value) == 1:
+                # Single-elements can be returned by themselves.
+                return next(iter(value))
+            else:
+                return frozenset(value)
+        elif isinstance(value, dict):
+            # Dict keys are always hashable. Can't be sure that values are
+            # hashable.
+            return cls._hashable(  # Reuse existing {list: tuple} logic.
+                sorted((k, cls._hashable(v)) for k, v in value.iteritems()))
+        else:
+            return value  # Any other type is assumed to already be hashable.
+
+    def __init__(self, answer_dict):
+        self.value = answer_dict
+        self._hashable_value = self._hashable(self.value['answer'])
+
+    def __hash__(self):
+        return hash(self._hashable_value)
+
+    def __eq__(self, other):
+        if isinstance(other, _HashableAnswer):
+            return self._hashable_value == other._hashable_value  # pylint: disable=protected-access
+        return False
+
+
+def _get_top_answers(answer_dicts_list, limit=None):
+    """Return the most frequent answers from the list of answer dicts.
+
+    Args:
+        answer_dicts_list: dict(*). The dict containing information about a
+            particular answer. Must contain the key 'answer', where it's value
+            is the actual answer's value. Other keys are ignored by the
+            calculation.
+        limit: int or None. Maximum number of elements to fetch. If None, the
+            entire colletion is returned.
```

Comment:
"collection" is a term of art in Oppia; maybe try expressing this another way.

Reply:
Done.